Финальная задание 5 спринта

karabasBaRaBaS-1276 opened the pull request:
Структуры Training и DaySteps реализуют интерфейс DataParser

rongo-base64 committed:
Финальная задание 5 спринта

ivannizh commented on internal/actioninfo/actioninfo.go:
```diff
@@ -1,9 +1,26 @@
+	for _, datastring := range dataset {
+		err := dp.Parse(datastring)
+		if err != nil {
+			log.Println(err)
```
При ошибке нужно переходить на следующую итерацию цикла, то есть добавить `continue`

ivannizh commented on internal/daysteps/daysteps.go:
```diff
@@ -1,13 +1,74 @@
+
+	steps, err := strconv.Atoi(parseData[0])
+	if err != nil {
+		return fmt.Errorf("%w: %w", errorFormateData, err)
```
Обернуть можно только одну ошибку, нужно или сделать `"%v: %w"` или использовать `errors.Join`

ivannizh reviewed (approved):
Роман, привет!

Отлично получилось поработать со структурами и интерфейсами, работа принята!

rongo-base64 committed:
fix: improve error handling in loop and fmt.Errorf usage

- Added 'continue' in for loop on non-nil error
- Replaced "%w: %w" with "%w: %v" in fmt.Errorf calls
